bin/embark

```diff
@@ -1,15 +1,1023 @@
+
+// -- package.json :: of an embark pkg -----------------------------------------
+
+function PkgJsonEmbark(filepath, kind) {
+  PkgJson.call(this, filepath);
+  this.kind = kind || 'invoked';
+  this.nodeRange = undefined;
+  this.pkg = undefined;
+  this.version = undefined;
+}
+setupProto(PkgJsonEmbark, PkgJson);
+
+PkgJsonEmbark.prototype.log = function () {
+  PkgJson.prototype.log.call(this);
+  this.logMissingVersion();
+  this.logUnsupportedNode();
+};
+
+PkgJsonEmbark.prototype.loglevel = 'error';
+
+PkgJsonEmbark.prototype.logMissingFile = function () {
+  if (PkgJson.prototype.logMissingFile.call(this)) {
+    reportMissingFile_PkgJsonEmbark(this.kind, this.loglevel);
+    exitWithError();
+  }
+};
+
+PkgJsonEmbark.prototype.logMissingVersion = function () {
+  var missing;
+  var oldlevel = this.loglevel;
+  this.loglevel = 'warn';
+  if (this.json && this.version === '???') {
+    missing = true;
+    reportMissingVersion(this.filepath, this.kind, this.loglevel);
+  }
+  this.loglevel = oldlevel;
+  return missing;
+};
+
+PkgJsonEmbark.prototype.logPkgErrors = function () {
+  if (PkgJson.prototype.logPkgErrors.call(this)) {
+    reportPkgErrors_PkgJsonEmbark(this.dirname, this.kind, this.loglevel);
+    exitWithError();
+  }
+};
+
+PkgJsonEmbark.prototype.logUnparsable = function () {
+  if (PkgJson.prototype.logUnparsable.call(this)) {
+    reportUnparsable_PkgJsonEmbark(this.kind, this.loglevel);
+    exitWithError();
+  }
+};
+
+PkgJsonEmbark.prototype.logUnsupportedNode = function () {
+  var missing;
+  var range = this.nodeRange;
+  if (typeof range === 'undefined') {
+    missing = true;
+    range = this.nodeRangeDefault;
+  }
+  var bad;
+  range = parseRange(range);
+  if (!range) {
+    bad = true;
+    range = this.nodeRangeDefault;
+  }
+  var procVer = semver.clean(process.version);
+  this.loglevel = 'error';
+  if (!semver.satisfies(procVer, range)) {
+    reportUnsupportedNode(
+      bad,
+      this.filepath,
+      this.kind,
+      this.loglevel,
+      missing,
+      this.nodeRangeDefault,
+      this.nodeRange,
+      this.pkg,
+      procVer,
+      range
+    );
+    exitWithError();
+  }
+};
+
+PkgJsonEmbark.prototype.noCheck = true;
+
+// if changing to the `nodeRangeDefault` value, make sure to manually check
+// that it's a valid semver range, otherwise fallback logic in the prototype
+// methods won't be reliable
+PkgJsonEmbark.prototype.nodeRangeDefault = semver.Range('>=8.11.3').range;
+
+PkgJsonEmbark.prototype.setNodeRange = function () {
+  if (isObject(this.json) &&
+      this.json.hasOwnProperty('engines') &&
+      this.json.engines.hasOwnProperty('node')) {
+    this.nodeRange = this.json.engines.node;
+  }
+};
+
+PkgJsonEmbark.prototype.setPkg = function () {
+  this.pkg = `embark@${this.version}`;
+};
+
+PkgJsonEmbark.prototype.setVersion = function () {
+  if (isObject(this.json) && this.json.version) {
+    this.version = this.json.version;
+  } else {
+    this.version = '???';
```

Comment:
is it intended?

Reply:
Yes, so that some shim messages will include `embark@???`. It's for the benefit of developers working on embark itself, a visual signal that something is wrong with package.json of invoked/local embark.